akka-remote/src/main/scala/akka/remote/PhiAccrualFailureDetector.scala

```diff
@@ -122,8 +134,11 @@ class PhiAccrualFailureDetector(
         // this is a known connection
         val interval = timestamp - latestTimestamp
         // don't use the first heartbeat after failure for the history, since a long pause will skew the stats
-        if (isAvailable(timestamp)) oldState.history :+ interval
-        else oldState.history
+        if (isAvailable(timestamp)) {
+          if (interval >= (acceptableHeartbeatPauseMillis / 2) && eventStream != null)
+            eventStream.publish(Warning(getClass.getName, getClass, s"heartbeat interval is growing too large: $interval millis"))
```

Comment:
Shouldn't the first one be this.toString or something instance specific, the usual thing is actor path, here we put the class name both as `logSource` and `logClass` which means it will be impossible to discern many different failure detectors logging warning from the same failure detector logging many warnings.